### DISCUSSION TITLE
Qwen2.5-7B-Instruct 在模型推理时，哪个token对应的hidden vector最能代表整个句子的语义。

### DISCUSSION THREAD

**Discussion by lilxmx** [QwenLM/Qwen3#1105] [General]
texts = tokenizer.apply_chat_template(prompts,tokenize=False,add_generation_prompt=True)
seq = tokenizer(texts, return_tensors="pt", padding=True, truncation=True).to("cuda:0")
init_seq_lens = seq["input_ids"].shape[1]
vector_outputs = model(**seq, output_hidden_states=True, return_dict=True)

我希望用model的encoder对文本进行embedding，请问seq的长度例如423。请问是最后一个token对应的hidden表示整句话的语义吗

**Comment by jklj077** [comment]
Qwen模型是生成式语言模型，model没有"encoder"，不应直接作为embedding模型使用。
